src/svc/layer.rs

```diff
@@ -0,0 +1,49 @@
+use std::marker::PhantomData;
+
+/// A stackable element.
+///
+/// Given a `Next`-typed inner value, produces a `Bound`-typed value.
+/// This is especially useful for composable types like `MakeClient`s.
+pub trait Layer<Next> {
```

Comment:
No, it's not really a config in this case.  The general idea is that we can use this to compose middlewares -- so that both the `Next` and `Bound` types implement `MakeClient`, for instance (see https://github.com/linkerd/linkerd2-proxy/blob/132be6bc31e7953cc55d7d2fa30d73e9bcbc51d1/src/telemetry/http/timestamp_request_open.rs#L64-L90).  `Layer` lets us implement a `MakeClient` in terms of another MakeClient`.

I agree that this isn't 100% obvious right now, but, from experience, I think this will be useful as we expand its usage.